docs/sampling.asciidoc

```diff
@@ -1,27 +1,77 @@
+all while still making it easy to find anomalous patterns in your applications, detect outages, track errors,
+and lower MTTR.
+
+Elastic APM supports two types of sampling:
+
+* <<head-based-sampling>>
+* <<tail-based-sampling>>
+
+[float]
+[[head-based-sampling]]
+==== Head-based sampling
+
+In head-based sampling, the sampling decision for each trace is made when that trace is initiated.
+Each trace has a defined and equal probability of being sampled.
 
 For example, a sampling value of `.2` indicates a transaction sample rate of `20%`.
 This means that only `20%` of traces will send and retain all of their associated information.
 The remaining traces will drop contextual information to reduce the transfer and storage size of the trace.
 
+Head-based sampling is quick and easy to set up.
+Its downside is that it's entirely random -- good
+data might be discarded purely due to chance.
+
+See <<configure-head-based-sampling>> to get started.
+
+**Distributed tracing with head-based sampling**
+
+In a distributed trace, the sampling decision is still made when the trace is initiated.
+Each subsequent service respects the initial service's sampling decision, regardless of its configured sample rate;
+the result is a sampling percentage that matches the initiating service.
+
+In this example, `Service A` initiates four transactions and has sample rate of `.5` (`50%`).
+The sample rates of `Service B` and `Service C` are ignored.
+
+image::./images/dt-sampling-example-1.png[Distributed tracing and head based sampling example one]
+
+In this example, `Service A` initiates four transactions and has a sample rate of `1` (`100%`).
+Again, the sample rates of `Service B` and `Service C` are ignored.
+
+image::./images/dt-sampling-example-2.png[Distributed tracing and head based sampling example two]
+
 [float]
-==== Why sample?
+[[tail-based-sampling]]
+==== Tail-based sampling
 
-Distributed tracing can generate a substantial amount of data,
-and storage can be a concern for users running `100%` sampling -- especially as they scale.
+In tail-based sampling, the sampling decision for each trace is made after the trace has completed.
+This means all traces will be analyzed against a set of rules, or policies, which will determine the rate at which they are sampled.
 
-The goal of probability sampling is to provide you with a representative set of data that allows
-you to make statistical inferences about the entire group of data.
-In other words, in most cases, you can still find anomalous patterns in your applications, detect outages, track errors,
-and lower MTTR, even when sampling at less than `100%`.
+Tail-based sampling reduces the risk of discarding important data, because the sampling decision is only made _after_
+each trace has been analyzed.
+However, because traces are all initially observed,
+storage and transfer costs may be higher than with head-based sampling.
```

Comment:
Is this true? Are there other downsides?

Reply:
Kinda. There are pros and cons.

Cons:
 - more data sent from APM agents to APM Server
 - APM Server will use more CPU, memory, and disk (without TBS, APM Server won't hit the disk at all except for maybe swapping)

Pros (aside from the feature itself):
 - less data sent from APM Server to Elasticsearch

I think that pro is pretty important to call out. If you run APM Server close to the instrumented application, then the additional transfer cost can be reduced. Since TBS is implemented in APM Server, data is dropped at APM Server and so less data is transferred to Elasticsearch.

Reply:
Very neat! Thanks.